stopify-continuations/src/callcc/jumper.ts

```diff
@@ -87,20 +83,27 @@ function func(path: NodePath<Labeled<FunctionT>>, state: State): void {
   }
   const restoreLocals = path.node.localVars;
 
+  const frame = t.identifier('$frame');
+
+
   // We instrument every non-flat function to begin with a *restore block*
   // that is able to re-construct a saved stack frame. When the function is
   // invoked in restore mode, its formal arguments are already restored.
   // The restore block must restore the local variables and deal with
   // the *arguments* object. The arguments object is a real pain and hurts
   // performance. So, we avoid restoring it faithfully unless we are explicitly
   // configured to do so.
-  const restoreBlock: t.Statement[] = [ ];
-  // Restore all local variables. Creates the expression:
-  //     [local0, local1, ... ] = topStack.locals;
-  restoreBlock.push(
-    t.expressionStatement(t.assignmentExpression('=',
-      t.arrayPattern(restoreLocals), t.memberExpression(topOfRuntimeStack,
-        t.identifier('locals')))));
+  const restoreBlock = [
+    t.variableDeclaration('const', [t.variableDeclarator(frame, popRuntimeStack)]),
+    t.expressionStatement(t.assignmentExpression('=', target,
+      t.memberExpression(frame, t.identifier('index')))),
+  ];
+
+  if (restoreLocals.length > 0) {
```

Comment:
How often do you find that `restoreLocals.length === 0`?

Reply:
Anecdotally, I see it happen regularly when working with tests. 
I don't know exactly how often this happens in normal code.